### PR TITLE
Network filter

### LINKED ISSUE #2297: Provider connection error Error response from daemon: Invalid filter 'scope'
### Do you want to request a *feature* or report a *bug*?

A bug.

### What did you do?

I downloaded the [traefik_linux-amd64](https://github.com/containous/traefik/releases/download/v1.4.0/traefik_linux-amd64) and built a Docker image. Afterwards, I created a Docker service using the newly built Docker image with traefik v1.4.0 on a Docker swarm mode cluster running RHEL Atomic Host v7.4.0 with Docker engine v1.13.1.

### What did you expect to see?

I should be seeing the web UI of the backend services running with appropriate traefik labels via the Docker swarm mode backend.

### What did you see instead?

I am seeing the `404 page not found` page.

And the logs:
```console
$ tail -f /var/log/traefik/traefik.log
time="2017-10-21T19:46:23+08:00" level=error msg="Provider connection error Error response from daemon: Invalid filter 'scope', retrying in 3.546136256s"
time="2017-10-21T19:46:27+08:00" level=error msg="Failed to list services for docker swarm mode, error Error response from daemon: Invalid filter 'scope'"
```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
$ traefik version
Version:      v1.4.0
Codename:     roquefort
Go version:   go1.9.1
Built:        2017-10-16_04:52:07PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

- Operating System: RHEL Atomic Host v7.4.0
- Docker Engine: v1.13.1-21.1.gitcd75c68.el7
- Orchestration: Docker swarm mode
`$ cat /etc/traefik/traefik.toml`
```toml
checkNewVersion = false
traefikLogsFile = "/var/log/traefik/traefik.log"
logLevel = "INFO"
InsecureSkipVerify = true
RootCAs = [ "/etc/ssl/ca.crt" ]
defaultEntryPoints = ["http","https"]

[accessLog]
  filePath = "/var/log/traefik/access.log"
  format = "json"

[retry]
  attempts = 3

[healthcheck]
  interval = "30s"

[web]
  address = ":8080"
  CertFile = "/etc/ssl/wildcard.crt"
  KeyFile = "/etc/ssl/wildcard.key"
    [web.auth.basic]
      users = ["username:$apr1$awbZvi2u$0tX%3IDelob/Hj4xD9u0H1"]
    [web.statistics]
      RecentErrors = 10
    [web.metrics]
      [web.metrics.prometheus]
        Buckets=[0.1,0.3,1.2,5.0]

[entryPoints]
  [entryPoints.http]
    address = ":80"
    [entryPoints.http.redirect]
       entryPoint = "https"
  [entryPoints.https]
    address = ":443"
    [entryPoints.https.tls]
      [[entryPoints.https.tls.certificates]]
        certFile = "/etc/ssl/wildcard.crt"
        keyFile = "/etc/ssl/wildcard.key"

[docker]
  endpoint = "tcp://master01.example.com:2376"
  watch = true
  domain = "example.com"
  swarmmode = true
  usebindportip = true
  exposedbydefault = true

  [docker.tls]
    ca = "/etc/ssl/docker/ca.pem"
    cert = "/etc/ssl/docker/client-cert.pem"
    key = "/etc/ssl/docker/client-key.pem"
```

### PR DESCRIPTION
**Pull request by ldez**
### Description

Regression from #2244

Fixes #2297

### CODE REVIEW

File: provider/docker/docker.go [hunk 1]
```diff
@@ -827,8 +827,16 @@ func (p *Provider) listServices(ctx context.Context, dockerClient client.APIClie
 	if err != nil {
 		return []dockerData{}, err
 	}
+
+	serverVersion, err := dockerClient.ServerVersion(ctx)
+
 	networkListArgs := filters.NewArgs()
-	networkListArgs.Add("scope", "swarm")
+	// https://docs.docker.com/engine/api/v1.29/#tag/Network (Docker 17.06)
+	if hasRequiredAPIVersion(serverVersion.APIVersion, "1.29", true) {
```
Comment (vdemeester) [inline provider/docker/docker.go]: Reading this, it makes me thinks it's only for `1.29`, where it's on `1.29` and above right ?

File: provider/docker/docker_test.go [hunk 2]
```diff
@@ -1091,3 +1091,72 @@ func TestDockerHasStickinessLabel(t *testing.T) {
...
+			fallbackState:   false,
+			expectedState:   false,
+		},
+	}
+
+	for _, test := range testCases {
+		test := test
+		t.Run(test.desc, func(t *testing.T) {
+
+			state := hasRequiredAPIVersion2(test.baseVersion, test.requiredVersion, test.fallbackState)
```
Comment (vdemeester) [inline provider/docker/docker_test.go]: typo ? (`hasRequireAPIVersion2` ?)

Comment (vdemeester) [reply provider/docker/docker_test.go]: (and other question, it's not exported, should we really test that ? :stuck_out_tongue:)

Comment (vdemeester) [reply provider/docker/docker_test.go]: Ah, and there is https://github.com/moby/moby/blob/master/api/types/versions/compare.go too :angel:

File: provider/docker/docker.go (hunk 1 above)
Comment (ldez) [reply provider/docker/docker.go]: it's >1.29

**Comment by nmengin** [comment]
@vdemeester 
The original PR #2244 was created to check `scope` instead of `driver`, I guess this PR just fixes probelm dues to old versions.

**Comment by danieladams456** [comment]
@ldez Thanks for the fix!  When I opened #2275 right when 1.4.0 came out I assumed it was a wontfix since I was on an old docker version, so I closed it prematurely.  Then I see it got fixed as a P0 in #2297.  I had been using the RC series for the json access logs -> stdout -> cloudwatch, so this lets me get to the GA series.

**Pull request merged**